diff -Nur plague-0.4.5.7-20081216/common/SSLConnection.py plague-0.4.5.7-20090612/common/SSLConnection.py
--- plague-0.4.5.7-20081216/common/SSLConnection.py	2009-07-17 15:40:04.000000000 +0200
+++ plague-0.4.5.7-20090612/common/SSLConnection.py	2009-06-12 19:06:38.000000000 +0200
@@ -81,7 +81,6 @@
             return
         self.__dict__["close_refcount"] = self.__dict__["close_refcount"] - 1
         if self.__dict__["close_refcount"] == 0:
-            self.shutdown()
             self.__dict__["conn"].close()
             self.__dict__["closed"] = True
 
@@ -144,6 +143,9 @@
                 return None
             except SSL.WantReadError:
                 time.sleep(0.2)
+            except SSL.SysCallError, (e, err):
+                time.sleep(0.2)
+                return None
         return None
 
 class PlgFileObject(socket._fileobject):
diff -Nur plague-0.4.5.7-20081216/ChangeLog plague-0.4.5.7-20090612/ChangeLog
--- plague-0.4.5.7-20081216/ChangeLog	2009-07-17 15:40:04.000000000 +0200
+++ plague-0.4.5.7-20090612/ChangeLog	2009-06-12 19:28:26.000000000 +0200
@@ -1,3 +1,11 @@
+2009-06-12  Michael Schwendt <mschwendt@fedoraproject.org>
+
+	* common/SSLConnection.py
+	  - Catch SSL.SysCallError exceptions for unexpected EOF.
+	  - Remove superfluous connection shutdown() call in close()
+	    because it results in an SSL.Error "uninitialized" exception
+	    with pyOpenSSL 0.7 and Python 2.6
+
 2008-12-16  Michael Schwendt <mschwendt@fedoraproject.org>
 
 	* common/SSLCommon.py
