--- common/daemonize.py.orig	2006-09-14 09:42:31.000000000 -0500
+++ common/daemonize.py	2006-09-14 09:41:34.000000000 -0500
@@ -69,7 +69,7 @@
          # to do this could make a filesystem unmountable.
          os.chdir("/")
          # Give the child complete control over permissions.
-         os.umask(0)
+         os.umask(0002)
       else:
          os._exit(0)      # Exit parent (the first child) of the second child.
    else:
