diff -Nur plague-0.4.5.3-orig/utils/user-manager.py plague-0.4.5.3/utils/user-manager.py
--- plague-0.4.5.3-orig/utils/user-manager.py	2008-09-05 11:17:27.000000000 +0200
+++ plague-0.4.5.3/utils/user-manager.py	2008-09-07 12:34:59.000000000 +0200
@@ -147,6 +147,7 @@
         if not item:
             raise UserManagerException("User %s doesn't exist." % email)
 
+        item = ResultSet(item, self.curs.description)
         own_jobs = item['own_jobs']
         job_admin = item['job_admin']
         user_admin = item['user_admin']
